swift/ql/lib/codeql/swift/dataflow/internal/DataFlowPrivate.qll

```diff
@@ -396,6 +419,13 @@ private module ParameterNodes {
     override ParamDecl getParameter() { result = param }
   }
 
+  class ClosureSelfParameterNode extends ParameterNodeImpl, ClosureSelfReferenceNode {
```

Comment:
This looks wrong, but I might be misinterpreting some detail of the Swift AST. I'd guess that a `ClosureExpr` corresponds roughly to a `LambdaExpr` in Java, which means that this node belongs to the surrounding scope of the closure and not the scope of the closure, which contradicts it being a parameter of the closure.

Reply:
Oops you're right. This doesn't make sense at all 👀

Reply:
Oh yes this doesn't make any sense!

Reply:
Fixed in 2465cc20f0108a9b80c85e438695cb1f43b2f3eb.